privacy/scrubber.go

```diff
@@ -81,6 +82,68 @@ func NewScrubber() Scrubber {
 	return scrubber{}
 }
 
+func (scrubber) ScrubRequest(bidRequest *openrtb2.BidRequest, enforcement Enforcement) *openrtb2.BidRequest {
+	if enforcement.UFPD {
+		// transmitUfpd covers user.ext.data, user.data, user.id, user.buyeruid, user.yob, user.gender, user.keywords, user.kwarray
+		// and device.{ifa, macsha1, macmd5, dpidsha1, dpidmd5, didsha1, didmd5}
+		if bidRequest.Device != nil {
+			bidRequest.Device.DIDMD5 = ""
+			bidRequest.Device.DIDSHA1 = ""
+			bidRequest.Device.DPIDMD5 = ""
+			bidRequest.Device.DPIDSHA1 = ""
+			bidRequest.Device.IFA = ""
+			bidRequest.Device.MACMD5 = ""
+			bidRequest.Device.MACSHA1 = ""
+		}
+		if bidRequest.User != nil {
+			bidRequest.User.Data = nil
+			bidRequest.User.Ext = scrubExtIDs(bidRequest.User.Ext, "data")
+			bidRequest.User.ID = ""
+			bidRequest.User.BuyerUID = ""
+			bidRequest.User.Yob = 0
+			bidRequest.User.Gender = ""
+			bidRequest.User.Keywords = ""
+			bidRequest.User.KwArray = nil
+		}
+	}
+	if enforcement.Eids {
+		//transmitEids covers user.eids and user.ext.eids
+		if bidRequest.User != nil {
+			bidRequest.User.EIDs = nil
+			bidRequest.User.Ext = scrubExtIDs(bidRequest.User.Ext, "eids")
```

Comment:
We could unmarshal `user.ext` at the start of this method if either UFPD or Eids are true, modify as necessary throughout the function, and at the end marshal it back once more if either UFPD or Eids are true. That would bring us down to at most 1 unmarshal/marshal pair for each ext. Thoughts?

Reply:
Done. Please see this change in this commit https://github.com/prebid/prebid-server/pull/2906/commits/bf4b27bc134953f5e85716623c8d550aac8471a5